wiki/Developer-Documentation.md

```diff
@@ -109,205 +93,70 @@ application is as follows:
    client library will fail-over to another chunk server that has the data; this
    fail-over is transparent to the application.
 
-Compiling C++ Side
+Compiling the C++ Code
 ------------------
-Compiling the C++ side produces metaserver and chunk server binaries, client and
-admin tools, and the C++ client library. We use **`cmake`** to build the C++
-code. Here are the steps:
-
-1. Change directory to the build root (i.e. `~/code/qfs/build/release`).
-
-        $ cd ~/code/qfs && mkdir -p build/release && cd build/release
-
-1. Generate the makefiles.
-
-        $ cmake ~/code/qfs/
-
-**Note:** If you have non-default locations for your boost or java files, then
-set `BOOST_ROOT`, `BOOST_LIBRARYDIR`, and `JAVA_INCLUDE_PATH` accordingly.
-
-        $ cmake -D BOOST_ROOT=/usr/local/include/boost-1_37 \\
-                -D BOOST_LIBRARYDIR=/usr/local/lib \\
-                -D JAVA_INCLUDE_PATH=/usr/java/default/include \\
-                ~/code/qfs
-
-1. Compile the source
-
-        $ make
-
-1. Once compilation is success, install the binaries. **Note:** Although the
-   following command suggests system installation what it actually does is to
-   just copy the binaries and libraries to separate directories under the build
-   directory.
-
-        $ make install
-
-At the end of the last step, you will have the following hierarchy:
-
-| Directory                    | Notes                          |
-| ---------                    | -----                          |
-|`~/code/qfs/build/release/bin`| This will contain the binaries.|
-|`~/code/qfs/build/release/bin/tools/`| This will contain the various QFS utilities that are equivalent to the *nix commands such as, cp, mv, mkdir, rmdir etc.|
-|`~/code/qfs/build/release/bin/tests/`|This contains simple unit-test programs to make/remove directories, read/write files, etc.|
-|`~/code/qfs/build/release/lib/`| This will contain the shared libraries.|
-|`~/code/qfs/build/release/lib/static/`| This will contain the static libraries.|
+Compiling the C++ code produces the metaserver, chunkserver, client, and admin
+tool binaries. It also produces the C++ client library. We use **`cmake`** to
+build the C++ code. You can use the top-level Makefile as a wrapper around cmake
+to build the project by simply executing `make`.
 
-**Note:** If you run into JNI build error on Cygwin due to gcc not knowing about
```

Comment:
I think we should keep this kind of troubleshoot information for known issues, but it's up to you.